src/state/slices/opportunitiesSlice/resolvers/uniV2/index.ts

```diff
@@ -1,159 +1,237 @@
 import { ethAssetId, ethChainId, fromAccountId, fromAssetId, toAssetId } from '@shapeshiftoss/caip'
 import type { MarketData } from '@shapeshiftoss/types'
-import { HistoryTimeframe } from '@shapeshiftoss/types'
-import type { ETH_FOX_POOL_CONTRACT_ADDRESS } from 'contracts/constants'
+import type { TokenAmount } from '@uniswap/sdk'
 import { WETH_TOKEN_CONTRACT_ADDRESS } from 'contracts/constants'
-import { fetchUniV2PairData, getOrCreateContract } from 'contracts/contractManager'
+import { fetchUniV2PairData, getOrCreateContractByType } from 'contracts/contractManager'
+import { ContractType } from 'contracts/types'
 import { ethers } from 'ethers'
 import { DefiProvider, DefiType } from 'features/defi/contexts/DefiManagerProvider/DefiCommon'
+import memoize from 'lodash/memoize'
+import type { BN } from 'lib/bignumber/bignumber'
 import { bn, bnOrZero } from 'lib/bignumber/bignumber'
 import { getEthersProvider } from 'lib/ethersProviderSingleton'
 import { toBaseUnit } from 'lib/math'
+import { selectZapperFullfilled } from 'state/apis/zapper/selectors'
+import { zapperApi } from 'state/apis/zapper/zapperApi'
 import type { ReduxState } from 'state/reducer'
 import type { AssetsState } from 'state/slices/assetsSlice/assetsSlice'
 import { marketData } from 'state/slices/marketDataSlice/marketDataSlice'
 import type { PortfolioAccountBalancesById } from 'state/slices/portfolioSlice/portfolioSliceCommon'
 import { selectPortfolioLoadingStatusGranular } from 'state/slices/portfolioSlice/selectors'
 import {
+  selectFeatureFlags,
   selectMarketDataById,
   selectPortfolioAccountBalancesBaseUnit,
 } from 'state/slices/selectors'
 
 import { foxEthLpAssetIds } from '../../constants'
-import type { GetOpportunityIdsOutput, GetOpportunityMetadataOutput } from '../../types'
-import type { OpportunityMetadataResolverInput, OpportunityUserDataResolverInput } from '../types'
+import type {
+  GetOpportunityIdsOutput,
+  GetOpportunityMetadataOutput,
+  LpId,
+  OpportunityMetadata,
+} from '../../types'
+import type {
+  OpportunitiesMetadataResolverInput,
+  OpportunityIdsResolverInput,
+  OpportunityUserDataResolverInput,
+} from '../types'
 import { calculateAPRFromToken0 } from './utils'
 
-export const uniV2LpMetadataResolver = async ({
-  opportunityId,
+const ethersProvider = getEthersProvider()
+
+const getBlockNumber = memoize(async () => await ethersProvider.getBlockNumber())
```

Comment:
ser

Reply:
That's actually on purpose, this is a memoized variant just for this module, since this is only used for metadata and would be fired ~700 times otherwise - note this is going to go away once we turn the flag on and fetch all from Zapper instead of making JSON-RPC requests